v2/internal/metrics/arm_client_metrics.go

```diff
@@ -60,16 +51,16 @@ func (a *ARMClientMetrics) RegisterMetrics() {
 }
 
 // RecordAzureSuccessRequestsTotal records the total successful number requests to ARM by increasing the counter.
-func (a *ARMClientMetrics) RecordAzureSuccessRequestsTotal(resourceName string, statusCode int, method HTTPMethod) {
+func (a *ARMClientMetrics) RecordAzureSuccessRequestsTotal(resourceName string, statusCode int, method string) {
```

Comment:
Why go for the less well typed parameter?

Reply:
My reasoning here is that it's what the actual `http.Request` type has for `Method`, and so when writing the `metrics_policy` if we had an enum, I would need to write an:
```
if method == http.PUT {
    typedMethod = HTTPPut
}
...
```

or a switch of the same.

That felt more awkward than just doing `method := raw.Method`. Not that we couldn't do the above but if the Go stdlib authors don't care to do it, it felt awkward for us to bolt it on above as it made the interface between us and HTTP more awkward.

Reply:
Fair enough.